### PR TITLE
Add SDL12COMPAT_USE_KEYBOARD_LAYOUT hint

### PR DESCRIPTION
**Pull request by sulix**
tl;dr: This change makes the behaviour change in #97 optional, and configured at runtime with the new ``SDL12COMPAT_USE_KEYBOARD_LAYOUT`` environment variable. The default is the current behaviour of always using the scancodes/US layout, and the user's current layout can be used by setting said variable to '1'.

SDL 1.2 seems to only account for (non-en_US) keyboard layouts on some platforms and configurations. In particular, the KeySym may contain the translated key, or may contain a "scancode" which is layout-independent.

In particular, on X11 the configured layout is used, but only (it seems) if it is the only configured layout. (On systems where multiple layouts are configured, the US layout is used regardless of the configured one.) Similarly, the windib backend deliberately forces the US layout in order to match the DirectX backend, which only provides scancodes.

Since this behaviour is very inconsistant between different configurations, this patch provides a way of configuring which is used at runtime:

If the ``SDL12COMPAT_USE_KEYBOARD_LAYOUT`` environment variable is set to a nonzero value, sdl12compat will translate keyboard input according to the current layout. This matches SDL 1.2 on X11 with only one configured
layout. Otherwise (the default), sdl12compat will use SDL2 scancodes for keyboard input, essentially forcing all input to the US layout. This matches SDL 1.2 on Windows, and on X11 with multiple configured layouts.

Regardless of the value of this environment variable, the 'unicode' value will be affected by keyboard layout.

For discussion and more details see:
- [sdl12compat issue #135: Non-US Keyboard layouts not supported](https://github.com/libsdl-org/sdl12-compat/issues/135)
- [sdl12compat PR #97: Use Scancodes instead of Keycodes](https://github.com/libsdl-org/sdl12-compat/pull/97)
- [SDL 1.2 issue #95: windib returns locale dependent SDLK_ codes](https://github.com/libsdl-org/SDL-1.2/issues/95)

### CODE REVIEW

File: src/SDL12_compat.c [hunk 1]
```diff
@@ -3108,7 +3112,11 @@ EventFilter20to12(void *data, SDL_Event *event20)
                 return 1;  /* ignore 2.0-style key repeat events */
             }
 
-            PendingKeydownEvent.key.keysym.sym = KeysymFromSDL2(event20);
+            if (TranslateKeyboardLayout) {
+                PendingKeydownEvent.key.keysym.sym = Keysym20to12(event20->key.keysym.sym);
+            } else {
+		        PendingKeydownEvent.key.keysym.sym = Scancode20toKeysym12(event20->key.keysym.scancode);
```
Comment (slouken) [inline src/SDL12_compat.c:3118]: Can you fix the whitespace here? Otherwise it looks good!
